### ISSUE TITLE
Cohort export is broken - filter appears to be applied at series or instance level, not case level

### ISSUE THREAD

**Issue by fedorov** [ImagingDataCommons/IDC-WebApp#595]
TL;DR: "I select the filter as in screenshot; I export the cohort; I look at the cohort in BQ, and it does not include anything other than SEG series"

Issue reproducible after "clear cache and hard reload"

https://imaging.datacommons.cancer.gov/
![image](https://user-images.githubusercontent.com/313942/114948587-4b547b00-9e1d-11eb-9934-6c8be75cc868.png)

As expected, cohort summary is showing CT, SEG and SR modalities.

![image](https://user-images.githubusercontent.com/313942/114948690-85be1800-9e1d-11eb-9f33-b0a9d924465f.png)

However, when cohort is exported, it appears that only SEG series are included in the cohort:

```sql
SELECT
  DISTINCT(modality)
FROM
  `canceridc-user-data.user_manifests.manifest_cohort_1_20210415_224921` AS my_cohort
JOIN
  `canceridc-data.idc_views.dicom_all` AS all_data
ON
  my_cohort.SOPInstanceUID = all_data.SOPInstanceUID
```

![image](https://user-images.githubusercontent.com/313942/114948740-9cfd0580-9e1d-11eb-8134-7f7ed6489049.png)

The problem is also apparent from the number of rows in the exported cohort manifest table:

![image](https://user-images.githubusercontent.com/313942/114948779-baca6a80-9e1d-11eb-9c89-6fc3cd06802b.png)

When I select the complete list of cases/studies I get 180 series (and each CT series is expected to have a lot of instances).

![image](https://user-images.githubusercontent.com/313942/114948942-067d1400-9e1e-11eb-8aa1-42eb21dffe01.png)

**Comment by s-paquette** [comment]
I've fixed the issue with derived filters not properly exporting all of a study. However, we need to decide HOW we want study level inclusion to work. There's two main ways:

1. When we say the records are included at the study level, we mean ALL of the records from the study; as a result, even records which don't match the filter come along. (In effect, the filter is used to determine studies *containing* instances with a given derived data value, and then *everything* from those studies is exported.
2. When we say the records are included at the study level, we mean only the records which match the filter OR which lack a value for that specific filter (i.e. they're NULL).

I wrote the code to do #1, but do we want this to be behaving as #2? Or should it (in theory) not matter?

@G-White-ISB There seems to be something wrong with the counts in the Search Scope/Collections area for this specific cohort filter. It shows only 3 collections, but there should be 4, and 4 will show on the collections table listing to the right.

**Comment by fedorov** [comment]
@s-paquette thank you.

> 1. [...] the filter is used to determine studies containing instances with a given derived data value, and then everything from those studies is exported.

That's how we've been defining the cohort all along, see https://learn.canceridc.dev/portal/data-exploration-and-cohorts#understanding-counts-in-the-search-results. 

> 2. When we say the records are included at the study level, we mean only the records which match the filter OR which lack a value for that specific filter (i.e. they're NULL).

No! We should not include cases that have records, which lack a value for that specific filter!

Did you figure out why this regression happened? I am confident export was working in production at some point.

**Comment by wlongabaugh** [comment]
@s-paquette, what is the use case for #2? Trying to wrap my head around this.

**Comment by s-paquette** [comment]
@fedorov 

> No! We should not include cases that have records, which lack a value for that specific filter!

I'm speaking of the study level, not the case level. The difference between #1 and #2 is:

For #1, all instances from the studies identified are included. 
For #2, only instances which have a matching value or which have NO value are included from that study. 

Let's say we have 2 studies, A and B. A has instances which match a given filter, or which are NULL. B has instances with values that match, values tat do NOT match, or NULL.

In use case #1, all instances of study A and B are included.
In use case #2, all instances in study A but *only* the NULL and matching instance in study B are included.

I have no idea if studies as I describe for Study B exist; if not, then use case #1 is the only use case which will occur. If, however, there are studies as I describe in study B, then use case #1 and use case #2 produce different manifests. @wlongabaugh I don't know what the use case would be, I'm just describing the two ways we can go about this, and want to know which one we should be doing.

**Comment by fedorov** [comment]
@s-paquette I think the reason I am confused is because this bug is about production tier, and in production tier we only have cohort defined at case level. The topic you are raising is important for introducing study- and series-level filters, but I thought it is different from fixing the bug we currently have in production.

Use case `#1` is appropriate if we apply filtering at the study level, as this will be consistent with our approach to filtering at the case level.

Use case `#2` would be appropriate if we apply filtering at the series level. 

We will need to have a separate discussion about NULLs.

**Comment by madelyngreyes** [comment]
<img width="1097" alt="Screen Shot 2021-04-22 at 9 50 57 AM" src="https://user-images.githubusercontent.com/16824894/115725954-46795500-a350-11eb-92a4-f172042c9dfd.png">

Seems to be working on the test tier as expected again.

**Comment by madelyngreyes** [comment]
Example of a cohort created on the production tier, exported, and then queried by modality, 
<img width="691" alt="Screen Shot 2021-04-29 at 1 23 05 PM" src="https://user-images.githubusercontent.com/16824894/116593051-acd11b00-a8ee-11eb-9672-43c6e9b1d153.png">
